enhancements/assets-generation/cloud-foundry-application-discovery/README.md

```diff
@@ -74,40 +74,233 @@ to implement the design.  For instance,
+`rake custom:start` command will be used.
+
+Some buildpacks, such as Python, that work on a variety of frameworks, do not 
+attempt to provide a default start command. For these cases, a Procfile should
+be used to specify any necessary commands for the app.
+
+###### Web process
+
+`web` is a [special process type](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#web-process-type)
+that is required for all applications. The `web` `PROCESS_TYPE` must be specified
+by either the buildpack or the Procfile.
+
+###### Specifying processes in manifest files
+
+Custom process types can also be configured via a manifest file. Read more about
+[manifests](https://v3-apidocs.cloudfoundry.org/version/3.163.0/#manifests).
+It is not recommended to specify processes in both a manifest and a Procfile for
+the same app.
+
+#### Route-level configuration
+
+This [configuration](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#routes)
+is for *creating* mappings between the app and a route. Each route is created if
+it does not already exist. The protocol will be updated for any existing route
+mapping.
+
+Example:
+
+```
+---
+  ...
+  routes:
+  - route: example.com
+    protocol: http2
+  - route: www.example.com/foo
+  - route: tcp-example.com:1234
+```
+
+##### Definition
+
+| Name | Type | Description |
+| ----- | ----- | ----- |
+| **route** | *string* | **(Required)** The route URI |
+| **protocol** | *string* | (Optional) Protocol to use for this route. Valid protocols are `http1`, `http2`, and `tcp`. |
+
+#### Service-level configuration
+
+This configuration is *creating* new service bindings between the app and a 
+service instance. The `services` field can take either an array of service 
+instance name strings or an array of the following service-level fields.
+
+##### Definition
+
+| Name | Type | Description |
+| ----- | ----- | ----- |
+| **name** | *string* | **(Required)** The name of the service instance to be bound to |
+| **binding\_name** | *string* | The name of the service binding to be created |
+| **parameters** | *object* | A map of arbitrary key/value pairs to send to the service broker during binding |
+
+#### Sidecar-level configuration
+
+This configuration is for the individual sidecar. Each sidecar is created if
+it does not already exist.
+
+##### Definition
+
+| Name | Type | Description |
+| ----- | ----- | ----- |
+| **name** | *string* | **(Required)** The identifier for the sidecars to be configured |
+| **command** | *string* | The command used to start the sidecar |
+| **process\_types** | *list of strings* | List of processes to associate sidecar with |
+| **memory** | *integer* | Memory in MB that the sidecar will be allocated |
 
-The following table depicts the relationship between the Cloud Foundry Application manifest fields and the proposed location in the canonical form manifest.
+### Proposal Specification
 
```

Comment:
```The following sections describe the proposed specification that will capture the CF application manifest into a canonical form during the discovery phase or a M2K operation.

These structures are intended to abstract the CF application manifest format so that changes to the CF manifest are contained.```